pkg/validating-webhooks/agentinstall/v1beta1/agent_admission_hook.go

```diff
@@ -0,0 +1,192 @@
+package v1beta1
+
+import (
+	"net/http"
+
+	"github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/openshift/assisted-service/models"
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	agentResource = "agents"
+
+	agentAdmissionGroup   = "admission.agentinstall.openshift.io"
+	agentAdmissionVersion = "v1"
+)
+
+// AgentValidatingAdmissionHook is a struct that is used to reference what code should be run by the generic-admission-server.
+type AgentValidatingAdmissionHook struct {
+	decoder *admission.Decoder
+}
+
+// NewAgentValidatingAdmissionHook constructs a new NewAgentValidatingAdmissionHook
+func NewAgentValidatingAdmissionHook(decoder *admission.Decoder) *AgentValidatingAdmissionHook {
+	return &AgentValidatingAdmissionHook{decoder: decoder}
+}
+
+// ValidatingResource is called by generic-admission-server on startup to register the returned REST resource through which the
+//                    webhook is accessed by the kube apiserver.
+// For example, generic-admission-server uses the data below to register the webhook on the REST resource "/apis/admission.agentinstall.openshift.io/v1/agentvalidators".
+//              When the kube apiserver calls this registered REST resource, the generic-admission-server calls the Validate() method below.
+func (a *AgentValidatingAdmissionHook) ValidatingResource() (plural schema.GroupVersionResource, singular string) {
+	log.WithFields(log.Fields{
+		"group":    agentAdmissionGroup,
+		"version":  agentAdmissionVersion,
+		"resource": "agentvvalidator",
```

Comment:
```suggestion
		"resource": "agentvalidator",
```
Maybe it should be a const?

Reply:
Fixed typo.
I did not add a constant for it, only two occurrences.